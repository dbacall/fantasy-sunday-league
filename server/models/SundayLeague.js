const mongoose = require('mongoose');
const Schema = mongoose.Schema;

const sundayLeagueSchema = new Schema({
  leagueName: String,
  owner: {
    type: Schema.Types.ObjectId,
    ref: 'User',
  },
});

module.exports = mongoose.model('SundayLeague', sundayLeagueSchema);
