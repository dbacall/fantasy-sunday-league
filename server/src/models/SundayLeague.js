const mongoose = require('mongoose');
const Schema = mongoose.Schema;

const sundayLeagueSchema = new Schema({
  name: String,
  owner: {
    type: Schema.Types.ObjectId,
    ref: 'users',
  },
});

// team ref

sundayLeagueSchema.virtual('teams', {
  ref: 'SundayLeagueTeam',
  localField: '_id',
  foreignField: 'league',
});

// season ref

sundayLeagueSchema.virtual('seasons', {
  ref: 'SundayLeagueSeason',
  localField: '_id',
  foreignField: 'league',
});

sundayLeagueSchema.set('toObject', { virtuals: true });
sundayLeagueSchema.set('toJSON', { virtuals: true });

module.exports = mongoose.model('SundayLeague', sundayLeagueSchema);
