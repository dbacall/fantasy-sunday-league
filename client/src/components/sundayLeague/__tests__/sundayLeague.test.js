import React, { Component } from 'react';
import { shallow } from 'enzyme';
import SundayLeague from '../sundayLeagueComponent';

describe('sundayLeague component tests:', () => {
  const props = { league: { leagueName: 'league1' } };
  const teams = [{ teamName: 'team1' }, { teamName: 'team2' }];

  it('should show the form when add team button clicked', () => {
    const wrap = shallow(<SundayLeague {...props} />);
    wrap.find('button').first().simulate('click');
    expect(wrap).toMatchSnapshot();
  });

  it('should show the name you type in to the input', () => {
    const wrap = shallow(<SundayLeague {...props} />);
    wrap.find('button').simulate('click');
    expect(wrap.find('input').prop('value')).toEqual('');
    wrap.find('input').simulate('change', {
      target: { value: 'team1' },
    });
    expect(wrap.find('input').prop('value')).toEqual('team1');
  });

  it('should submit the team', () => {
    // const submitTeam = jest.fn();
    // const wrap = shallow(<SundayLeague {...props} submitTeam={submitTeam} />);
    // wrap.find('button').simulate('click');
    // wrap.find('input').simulate('change', {
    //   target: { value: 'team1' },
    // });
    // wrap.find('button').at(1).simulate('click');
  });

  it('should show teams if in props', () => {
    const wrap = shallow(<SundayLeague {...props} teams={teams} />);
    expect(wrap).toMatchSnapshot();
  });
});
