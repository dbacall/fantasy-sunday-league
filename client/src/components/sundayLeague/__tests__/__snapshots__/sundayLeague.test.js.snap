// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sundayLeague component tests: should show teams if in props 1`] = `
<div>
  <h1>
    league1
  </h1>
  <button
    onClick={[Function]}
  >
    Add Team
  </button>
  <h2>
    Teams
  </h2>
  <div
    key="0"
  >
    <Link
      to={
        Object {
          "pathname": "/sunday-league/undefined",
          "state": Object {
            "league": Object {
              "leagueName": "league1",
            },
          },
        }
      }
    >
      team1
    </Link>
  </div>
  <div
    key="1"
  >
    <Link
      to={
        Object {
          "pathname": "/sunday-league/undefined",
          "state": Object {
            "league": Object {
              "leagueName": "league1",
            },
          },
        }
      }
    >
      team2
    </Link>
  </div>
</div>
`;

exports[`sundayLeague component tests: should show the form when add team button clicked 1`] = `
<div>
  <h1>
    league1
  </h1>
  <button
    onClick={[Function]}
  >
    Add Team
  </button>
  <div>
    <form
      onSubmit={[Function]}
    >
      <input
        className="add-team-input"
        onChange={[Function]}
        placeholder="Team name here..."
        type="text"
        value=""
      />
      <button
        className="team-submit-btn"
      >
        Add
      </button>
    </form>
  </div>
  <h2>
    Teams
  </h2>
</div>
`;
